### ISSUE TITLE
[RRFC] npm install uses system-wide node installation instead of local one

### ISSUE THREAD

**Issue by Osiris-Team** [npm/rfcs#531]
I was told to move this here. Previous issue: https://github.com/npm/cli/issues/4390
### Version

17.4.0

### Platform

Windows x64

### What steps will reproduce the bug?

1. Download node installation from here: https://nodejs.org/download/release/latest-v17.x/
2. Unpack the zip/tar somewhere
3. Create a new empty folder and open a terminal in that folder
4. Enter the following command in that terminal: `<npm-path> install puppeteer` (can be any module, puppeteer just a random example, also the npm-path must be pointing to the npm file from the unpacked node installation from before)
5. Voila.

### How often does it reproduce? Is there a required condition?

NodeJS must NOT be installed system-wide.

### What is the expected behavior?

It should check for an existing `node.exe` in its installation directory first, and only use the global `node` variable if there is no `node.exe` in its installation directory.

### What do you see instead?

```txt
...
npm ERR! path C:\WORK\node_modules\puppeteer
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node install.js
...
```
As you can see npm tries to execute the command `node install.js`, the problem is, that node was not installed system-wide and thus there is no global `node` variable available.

**Comment by ljharb** [comment]
If it’s not available on the command line in the PATH, how can it be found? The current directory isn’t part of the PATH on non-windows systems.

**Comment by Osiris-Team** [comment]
No. If you run `npm install` that will use the system wide nodejs installation like it should and currently does.
If you however use something like this `C:\Users\Desktop\nodejs-latest\npm` it should NOT use the system-wide node js installation, but the one in `C:\Users\Desktop\nodejs-latest\`. 
Hope this clarifies it.

**Comment by wraithgar** [comment]
Ok yes that kind of thing was what was explicitly not supported, for reasons stated previously.  If something else runs node it has no way of knowing that you started the process w/ another node.exe.  The only way to be certain the same version of node is ran is to let the operating system decide every time.

**Comment by wraithgar** [comment]
If you need some other node.exe to take precedence, you should prepend the directory it lives in to your `PATH` environment variable.  Even if you scope this to your current shell or call to npm/node that `PATH` will be inherited by subsequent shell calls.

**Comment by Osiris-Team** [comment]
@wraithgar Can you elaborate on the security risk. Because I am unable to see a security risk in my approach on fixing this flaw.

**Comment by Osiris-Team** [comment]
Btw this would also mean that the binary version of node js is not fully functional.

**Comment by wraithgar** [comment]
The security risk was in running `node.exe` from the cwd, which is not what you are suggesting.

The binary version of node.js is fully functional, but it has to be installed.  Being installed doesn't necessarily mean it lives in a system-level folder, but it does mean that it is accessible via the `PATH` environment variable.  The [node.js binary installation docs](https://github.com/nodejs/help/wiki/Installation) point this out, it is a prerequisite.

**Comment by Osiris-Team** [comment]
In my case however, I would not want to change the user's PATH variable thus I am suggesting this approach.

**Comment by wraithgar** [comment]
That's why you use `set` instead of `setx`.  `set` only changes it for the current shell context, not globally.

**Comment by Osiris-Team** [comment]
Going to give that a try and see if it works. Thanks!

**Comment by Wissperwind** [comment]
How can I install node system wide?
I have the same problem:
```
431 verbose Windows_NT 10.0.14393
432 verbose node v18.4.0
433 verbose npm  v8.12.1
434 error code 1
435 error path {some folder}
436 error command failed
437 error command C:\Windows\system32\cmd.exe /d /s /c node install.js
438 error Der Befehl "node" ist entweder falsch geschrieben oder
438 error konnte nicht gefunden werden.
439 verbose exit 1
```

I installed the normal node with the msi installer. I checked the path. In the system path, not the user path there is the node exe. In C:\Program Files\nodejs is the node exe and that is exactly what is written in system path.

But why do I get the error? I thought it would find the exe because it is in a folder listed in path...
